modules/ingest-grok/src/main/java/org/elasticsearch/ingest/processor/GrokProcessor.java

```diff
@@ -71,12 +68,23 @@ Grok getGrok() {
         return grok;
     }
 
-    public static class Factory implements Processor.Factory<GrokProcessor> {
+    public final static class Factory implements Processor.Factory<GrokProcessor> {
 
-        private final Path grokConfigDirectory;
+        private final static String[] PATTERN_NAMES = new String[] {
+            "aws", "bacula", "bro", "exim", "firewalls", "grok-patterns", "haproxy",
+            "java", "junos", "linux-syslog", "mcollective-patterns", "mongodb", "nagios",
+            "postgresql", "rails", "redis", "ruby"
+        };
+        private final Map<String, String> builtinPatternBank;
 
-        public Factory(Path configDirectory) {
-            this.grokConfigDirectory = configDirectory.resolve("ingest").resolve("grok");
+        public Factory() throws Exception {
+            Map<String, String> builtinPatterns = new HashMap<>();
+            for (String pattern : PATTERN_NAMES) {
+                try(InputStream is = getClass().getResourceAsStream("/patterns/" + pattern)) {
```

Comment:
thinking out loud: do we want to have patterns under a specific package? or is /patterns ok in the classpath?